src/open_prime_hunters_rando/entities/portal.py

```diff
@@ -0,0 +1,23 @@
+from open_prime_hunters_rando.entities.entity_type import EntityFile
+
+
+def patch_portals(parsed_file: EntityFile, portals: list, room_name: str) -> None:
+    for portal in portals:
+        entity_id = portal["entity_id"]
+        filename: str = portal["entity_filename"]
+
+        entity = EntityFile.get_entity(parsed_file, entity_id)
+
+        # target_index is the load_index of the desination portal
+        entity.data.target_index = portal["target_index"]
+
+        # filenames must have a length of 15
+        if len(filename) != 15:
+            raise ValueError(f"{filename} is an invalid length. Expected 15, recieved {len(filename)}!")
+
+        # filenames must end with certain characters
+        valid_endings = (".", ".b", ".bi")
+        if not filename.endswith(valid_endings):
+            raise ValueError(f"Invalid entity filename provided for entity {entity_id} in {room_name}!")
```

Comment:
rather than these two checks, perhaps you should take in the same value you have defined in `level_data` and validate it using that? so instead of passing in `unit3_Land_Ent.`, you pass in `unit3_Land_Ent`. then you can do something like this:
```py
ALL_ENTITY_FILES = {
    entity_file 
    for level_data in (ALINOS | CELESTIAL_ARCHIVES | VESPER_DEFENSE_OUTPOST | ARCTERRA | OUBLIETTE).values()
    if (entity_file := level_data.entity_file) is not None
} # in level_data.py
```
and then
```py
if filename not in ALL_ENTITY_FILES:
    raise ValueError(f"Invalid entity filename for portal entity {entity_id} in {room_name}: {filename}")
filename = f"{filename}.bin" # add the file extension 
filename = filename[:15] # truncate to 15 characters 
```

Reply:
I dont think just getting the name will work because a lot of the filenames for the portals don't match case with the entity file name. Could always add a new field to the level data file